src/Microsoft.OData.Core/ODataInputContext.cs

```diff
@@ -189,14 +189,8 @@ internal ODataSimplifiedOptions ODataSimplifiedOptions
         /// </summary>
         public void Dispose()
         {
-            if (this.disposed)
-            {
-                return;
-            }
-

```

Comment:
why remove this codes? It's unsafe?

Reply:
See https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1063?view=vs-2019 on how to implement IDisposable correctly.

The Dispose method should have the code below, and only the code below. Any additional cleanup must be performed by Dispose(bool disposing).

```csharp
        public void Dispose()
        {
            this.Dispose(true);
            GC.SuppressFinalize(this);
        }
```

Reply:
@gathogojr @g2mula What do you think?